app/src/components/common/Tooltip.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div
+    class="tooltip-box"
+    @mousemove="handleMouseMove"
+  >
+    <slot />
+    <div :class="['tooltip', styleType]">
+      <span :class="['text', styleType]">{{ text }}</span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Component from 'vue-class-component'
+import Vue from 'vue'
+import { Prop } from 'vue-property-decorator'
+import { Point, Size } from '@/types'
+
+@Component
+export default class Tooltip extends Vue {
+  @Prop({ required: true }) text!: string
+  position: Point = { x: 0, y: 0 }
+  tooltipBoxSize: Size = { width: 100, height: 50 } //max-height and max-width values
+
+  get styleType (): string {
+    if (this.position.y < this.tooltipBoxSize.height + 50) {
+      //taking into account margin (50px)
+      return 'bottom'
+    } else if (
+      this.position.x >
+      window.innerWidth - this.tooltipBoxSize.width + 20
+    ) {
+      //taking into account margin (20px)
+      return 'left'
+    } else if (this.position.x < this.tooltipBoxSize.width + 20) {
+      //taking into account margin (20px)
+      return 'right'
+    } else {
+      return 'top'
+    }
+  }
+
+  handleMouseMove (event: MouseEvent) {
+    this.position = { x: event.clientX, y: event.clientY }
+  }
+}
+</script>
+
+<style scoped>
+.tooltip-box {
+  max-width: 100px;
+  max-height: 50px;
```

Comment:
what if I need to show a long text in a tooltip?

Reply:
Just removed this constraint in the recent commit.